tortoise/tortoise.go

```diff
@@ -652,16 +653,19 @@ func (t *turtle) getBallotBeacon(ballot *types.Ballot, logger log.Log) ([]byte,
 
 // HandleIncomingLayer processes all layer ballot votes
 // returns the old pbase and new pbase after taking into account ballot votes.
-func (t *turtle) HandleIncomingLayer(ctx context.Context, layerID types.LayerID) error {
+func (t *turtle) HandleIncomingLayer(ctx context.Context, lid types.LayerID) error {
 	defer t.evict(ctx)
 
 	tctx := wrapContext(ctx)
 	// unconditionally set the layer to the last one that we have seen once tortoise or sync
 	// submits this layer. it doesn't matter if we fail to process it.
-	if t.Last.Before(layerID) {
-		t.Last = layerID
+	if t.Last.Before(lid) {
+		t.Last = lid
```

Comment:
why shouldn't t.Last get updated by the clock?

Reply:
i don't get the question. Last should be updated by the clock (but currently isn't), Processed shouldn't

Reply:
ok. the PR description made me think you were going to update Last by the clock. maybe leave a TODO?

Reply:
why? there is already an issue for that, and this change affects Last only minimally